src/ChainSafe.Gaming.WalletConnect/Models/WalletConnectWalletModel.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using ChainSafe.Gaming.Web3.Environment;
+using Newtonsoft.Json;
+using WalletConnectSharp.Common.Logging;
+using WalletConnectSharp.Sign.Models.Engine;
+
+namespace ChainSafe.Gaming.WalletConnect.Models
+{
+    public class WalletConnectWalletModel
+    {
+        [JsonProperty("name")]
+        public string Name { get; private set; }
+
+        [JsonProperty("mobile")]
+        public WalletLinkModel Mobile { get; private set; }
+
+        [JsonProperty("desktop")]
+        public WalletLinkModel Desktop { get; private set; }
+
+        [JsonProperty("image_url")]
+        public ImageUrlsModel Images { get; private set; }
+
+        public void OpenDeeplink(ConnectedData data, IOperatingSystemMediator operatingSystemMediator, bool useNative = false)
+        {
+            string uri = string.Empty;
+
+            if (operatingSystemMediator.IsMobilePlatform)
+            {
+                switch (operatingSystemMediator.Platform)
```

Comment:
You can move mobile handling logic in a separate method (or a local function) to reduce nesting.

Reply:
okay

Reply:
done